stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -187,63 +187,102 @@ column. In conjunction with [`nonzeros`](@ref) and
 nzrange(S::AbstractSparseMatrixCSC, col::Integer) = getcolptr(S)[col]:(getcolptr(S)[col+1]-1)
 nzrange(S::SparseMatrixCSCView, col::Integer) = nzrange(S.parent, S.indices[2][col])
 
+function isstored(A::AbstractSparseMatrixCSC, i::Integer, j::Integer)
+    rows = rowvals(A)
+    for istored in nzrange(A, j) # could do binary search if the row indices are sorted?
+        i == rows[istored] && return true
+    end
+    return false
+end
+
+Base.replace_in_print_matrix(A::AbstractSparseMatrix, i::Integer, j::Integer, s::AbstractString) =
+    isstored(A, i, j) ? s : Base.replace_with_centered_mark(s)
+
 function Base.show(io::IO, ::MIME"text/plain", S::AbstractSparseMatrixCSC)
     xnnz = nnz(S)
     m, n = size(S)
     print(io, m, "×", n, " ", typeof(S), " with ", xnnz, " stored ",
               xnnz == 1 ? "entry" : "entries")
-    if xnnz != 0
+    if !(m == 0 || n == 0)
         print(io, ":")
         show(IOContext(io, :typeinfo => eltype(S)), S)
     end
 end
 
 Base.show(io::IO, S::AbstractSparseMatrixCSC) = Base.show(convert(IOContext, io), S::AbstractSparseMatrixCSC)
-function Base.show(io::IOContext, S::AbstractSparseMatrixCSC)
-    nnz(S) == 0 && return show(io, MIME("text/plain"), S)
 
-    ioc = IOContext(io, :compact => true)
-    function _format_line(r, col, padr, padc)
-        print(ioc, "\n  [", rpad(rowvals(S)[r], padr), ", ", lpad(col, padc), "]  =  ")
-        if isassigned(nonzeros(S), Int(r))
-            show(ioc, nonzeros(S)[r])
-        else
-            print(ioc, Base.undef_ref_str)
-        end
+const brailleBlocks = UInt16['⠁', '⠂', '⠄', '⡀', '⠈', '⠐', '⠠', '⢀']
+function _show_with_braille_patterns(io::IOContext, S::AbstractSparseMatrixCSC)
+    m, n = size(S)
+    (m == 0 || n == 0) && return show(io, MIME("text/plain"), S)
+
+    # The maximal number of characters we allow to display the matrix
+    local maxHeight::Int
+    local maxWidth::Int
+    maxHeight = displaysize(io)[1] - 4 # -4 from [Prompt, header, newline after elements, new prompt]
+    maxWidth = displaysize(io)[2] ÷ 2
```

Comment:
Note that this can be shortened to
```jl
local maxHeight::Int , maxWidth::Int
```
or even
```jl
local maxHeight::Int , maxWidth::Int = displaysize(io)[1] - 4, displaysize(io)[2] ÷ 2
```